06_puppy/alextmz/storer_test.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type storerSuite struct {
+	suite.Suite
+	st Storer
+}
+
+const (
+	IDPuppyDoesNotExist    = 99999
+	IDPuppyInvalidNegative = -1
+)
+
+func (su *storerSuite) TestCreatePuppy() {
+	// can we create without error?
+	p1 := Puppy{Breed: "Dogo", Colour: "White", Value: 500}
+	// this value copy is here, not below, to guarantee that it is not going
+	// to be modified before being used on the relevant test
+	expected := p1
+	err := su.st.CreatePuppy(&p1)
+	su.NoError(err)
+	// do we error when creating something that already exists?
+	err = su.st.CreatePuppy(&p1)
+	su.NotNil(err)
+	// what we create and what we read back match?
+	actual, _ := su.st.ReadPuppy(p1.ID)
+	actual.ID = 0
+	su.Equal(expected, actual)
+	// do we error when trying to create a puppy from a nil pointer?
+	var p4 *Puppy
+	err = su.st.CreatePuppy(p4)
+	su.NotNil(err)
+	// do we error when trying to create an already identified Puppy?
+	p2 := Puppy{Breed: "Mastiff", Colour: "Brindle", Value: 700}
+	p2.ID = IDPuppyDoesNotExist
```

Comment:
personally i'd prefer the number inlined, or defined here.

Reply:
Any specific reason? They look good to me in being 2 constants that save 7 values from being hard-coded, but I agree that being globals is not ideal.

Reply:
nm - I ended up taking those off to make the tests more isolated, hope this was the reason :-)